### PR TITLE
Update Heroku docs

### LINKED ISSUE #11759: Update Heroku Documentation
Heroku documentation is out-of-date with respect to upgrading Metabase.

We no longer bump the `metabase-deploy` repository on new releases and instead update the `metabase-buildpack` repo.

We should update the Heroku documentation with respect to the following:
- Remove any old artifacts (e.g. https://github.com/metabase/metabase-heroku and related https://elements.heroku.com/buildpacks/metabase/metabase-heroku )
- Update the upgrade instructions
  - Both for upgrades using the "empty commit" method
  - As well as possibly using pipelines to promote upgrades of Metabase

### PR DESCRIPTION
**Pull request by dacort**
This fixes #11759 

- Cleans up the commands needed to upgrade in a standard deploy
- Adds information about Heroku pipelines in a CD environment
- Adds info about syncing databases between apps and pinning versions

### CODE REVIEW

**Review by mazameli — APPROVED** [review]
The subject matter is above my pay grade, but I didn't notice any typos or broken links (though I probably wouldn't capitalize `staging` or `production` since they're adjectives rather than proper nouns, but that's nit-picky). 

I'll leave it up to you if you think this should get an additional review from someone who's more familiar with Heroku, like Tom.

**Comment by dacort** [comment]
> I probably wouldn't capitalize `staging` or `production`
Yea, I debated about that for a bit and think you're right. Will tweak that.

@tlrobinson Feel free to take a look if you like. I tweaked our description of how we use buildpacks/metabase-deploy repository and then the bigger change is a section on how to use Heroku pipelines to test new versions of Metabase.

File: docs/operations-guide/running-metabase-on-heroku.md
Comment (salsakran) [inline docs/operations-guide/running-metabase-on-heroku.md]: I know this is how things actually work, but this seems really lame as an upgrade flow =(

**Review by salsakran — APPROVED** [review]
I haven't manually tested all of the command line examples,  but this matches my understanding of how this all works.

I'm not certain we want to provide that much real estate to pipelines, but I'd default to merging this as-is.

File: docs/operations-guide/running-metabase-on-heroku.md
Comment (dacort) [reply docs/operations-guide/running-metabase-on-heroku.md]: Agreed. I hunted around for a bit to see if there was a better way, but didn’t come across much. This is part of the reason I included the pipelines component - it at least provides a testable and continuous deployment mechanism. Something I would want were I running Metabase in production on Heroku. That said, it still requires the empty commit strategy.

**Pull request merged**
